include/range/v3/view/iota.hpp

```diff
@@ -524,16 +543,17 @@ namespace ranges
                 detail::take_exactly_view_<iota_view<Val>, true>>
             ints_fn::operator()(Val from, Val to) const
             {
-                return {iota_view<Val>{from}, detail::iota_minus_(to, from)};
+                return {iota_view<Val>{from}, detail::ints_distance_(from, to)};
             }
 
             struct closed_ints_fn
             {
                 template<typename Val,
                     CONCEPT_REQUIRES_(Integral<Val>())>
+                RANGES_DEPRECATED("view::closed_ints is deprecated; use view::closed_indices instead!")
                 detail::take_exactly_view_<iota_view<Val>, true> operator()(Val from, Val to) const
                 {
-                    return {iota_view<Val>{from}, detail::iota_minus_(to, from) + 1};
+                    return {iota_view<Val>{from}, detail::ints_distance_(from, to) + 1};
```

Comment:
@ericniebler I've split `ints_distance_` into `ints_open_distance_` and `ints_closed_distance`. 

Its not about "jus" doing `+1`, but rather, about making sure that this `+1` doesn't overflow the signed difference_type that `view::take_exactly` takes as argument. 

So for open ranges, one checks that:
- `from <= to` (for both signed and unsigned integers),
- `Unsigned(to - from) <= Unsigned(numeric_limits<difference_type>::max())` for unsigned integers only, otherwise overflowing the signed difference type is undefined behavior.

For closed ranges one has the same checks as for open ranges, plus:
- `open_distance < numeric_limits<difference_type>::max()` (for both)
since otherwise `open_distance + difference_type(1) will overflow, introducing undefined behavior. 

@CaseyCarter, does this makes sense? (or more sense as before?).

> What they can't do is get a closed range that covers the entire range of integers without fear of overflow.

For unsigned integers they can only cover a sub-range of (half - 1) size. I think its worth it to assert on this to prevent undefined behavior. 

> And I disapprove of the inconsistent naming, FWIW.

I've renamed `inclusive_indices` to `closed_indices` "for consistency", but I think that `inclusive` is a better name, and we are deprecating `closed_ints` anyways, so if we remove that at some point, we will be left with `closed_indices` only. I'd rather not make a breaking change there again to rename it to `inclusive_indices` if we happen to like that better. Idk, what do you guys like more? Are there better names?